### ISSUE TITLE
Conda uninstall enum34 broke conda

### ISSUE THREAD

**Issue by nswitanek** [conda/conda#7028]
**I'm submitting a...**
  - [X] bug report
  - [ ] feature request

### Steps to Reproduce

I was trying to upgrade python 2.7 to python 3.4. enum34 was giving me an Unsatisfiable Error:
```
conda install python=3.4
Fetching package metadata ...........
Solving package specifications: .

UnsatisfiableError: The following specifications were found to be in conflict:
  - enum34 -> python 2.7*
  - python 3.4*
Use "conda info <package>" to see the dependencies for each package.
```

So I tried uninstalling enum34
```
conda uninstall enum34
Fetching package metadata ...........
Solving package specifications: .

Package plan for package removal in environment /Users/ns186048/anaconda:

The following packages will be REMOVED:

    enum34: 1.1.2-py27_0

Proceed ([y]/n)? y
```

Then, assuming that I'd removed the troublesome dependency, I tried installing python 3.4.

```
$ conda install python=3.4
Traceback (most recent call last):
  File "/Users/user/anaconda/bin/conda", line 11, in <module>
    sys.exit(main())
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 181, in main
    from ..exceptions import conda_exception_handler
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 11, in <module>
    from ._vendor.auxlib.entity import EntityEncoder
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 243, in <module>
    from enum import Enum
ImportError: No module named enum
```

I determined that by removing enum34 I had broken conda generally...

```
$ conda
Traceback (most recent call last):
  File "/Users/user/anaconda/bin/conda", line 11, in <module>
    sys.exit(main())
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 181, in main
    from ..exceptions import conda_exception_handler
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 11, in <module>
    from ._vendor.auxlib.entity import EntityEncoder
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 243, in <module>
    from enum import Enum
ImportError: No module named enum
```
 So I tried re-installing enum34 using pip
```
$ pip install enum34
Requirement already satisfied: enum34 in /Users/ns186048/anaconda/lib/python2.7/site-packages
```
I also tried pip installing enum, since the ImportError was complaining about missing the enum module after I conda uninstalled enum34.
```
$ pip install enum
Collecting enum
  Downloading enum-0.4.6.tar.gz
Requirement already satisfied: setuptools in /Users/ns186048/anaconda/lib/python2.7/site-packages/setuptools-20.3-py2.7.egg (from enum)
Building wheels for collected packages: enum
  Running setup.py bdist_wheel for enum ... done
  Stored in directory: /Users/user/Library/Caches/pip/wheels/d9/b9/23/e7aa8b7d643b49a3df6532fddf18d7ba3863d706c836d59515
Successfully built enum
Installing collected packages: enum
Successfully installed enum-0.4.6
```
And that changed slightly the nature of the import error, but I don't seem to be closer to resolving the problem with conda I created by uninstalling enum34.
```
$ conda info
Traceback (most recent call last):
  File "/Users/user/anaconda/bin/conda", line 11, in <module>
    sys.exit(main())
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 182, in main
    return conda_exception_handler(_main, *args)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 644, in conda_exception_handler
    return handle_exception(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 634, in handle_exception
    print_unexpected_error_message(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 579, in print_unexpected_error_message
    from .base.context import context
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/base/context.py", line 21, in <module>
    from ..common.configuration import (Configuration, LoadError, MapParameter, PrimitiveParameter,
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/common/configuration.py", line 28, in <module>
    from enum import Enum, EnumMeta
ImportError: cannot import name EnumMeta
```

## Expected Behavior
I expected that I could remove the package enum34 that was offending my attempts to upgrade python and then proceed with my python upgrade.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
$ conda info
Traceback (most recent call last):
  File "/Users/user/anaconda/bin/conda", line 11, in <module>
    sys.exit(main())
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 182, in main
    return conda_exception_handler(_main, *args)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 644, in conda_exception_handler
    return handle_exception(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 634, in handle_exception
    print_unexpected_error_message(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 579, in print_unexpected_error_message
    from .base.context import context
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/base/context.py", line 21, in <module>
    from ..common.configuration import (Configuration, LoadError, MapParameter, PrimitiveParameter,
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/common/configuration.py", line 28, in <module>
    from enum import Enum, EnumMeta
ImportError: cannot import name EnumMeta
```
</p></details>

**Comment by nswitanek** [comment]
I installed anaconda3, which gives me Python 3.6, but now I don't have access to the conda environments I established with anaconda2:

```
$ conda info --envs
# conda environments:
#
                         /Users/user/anaconda/envs/magenta
                         /Users/user/anaconda/envs/pytorch
                         /Users/user/anaconda/envs/tensorflow
base                  *  /Users/user/anaconda3
$ source activate pytorch
Could not find conda environment: pytorch
You can list all discoverable environments with `conda info --envs`.
```
So now I think the easiest way forward is to figure out how to access my old environments, but I should be able to find a thread on that online somewhere.

The trouble referenced in this issue may still be worth addressing, but I personally am no longer stopped by it.

**Comment by kalefranz** [comment]
‘conda config --describe envs_dirs’

**Comment by kalefranz** [comment]
Conda should have never let you uninstall enum34 in the first place.  This was a bug that was fixed with conda 4.5.0.  We have some hard-coded solver behavior that typically prevents major version changes of python within environments.  It's been decided in the past that that's the desired behavior for conda.  See

https://github.com/conda/conda/blob/c69c57ae645d90f8187d88aafc5f622f4164861a/conda/core/solve.py#L286-L294

**Comment by zhangxue123** [comment]
pip install enum34 can solve this.


$ conda info
Traceback (most recent call last):
  File "/Users/user/anaconda/bin/conda", line 11, in <module>
    sys.exit(main())
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 182, in main
    return conda_exception_handler(_main, *args)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 644, in conda_exception_handler
    return handle_exception(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 634, in handle_exception
    print_unexpected_error_message(e)
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 579, in print_unexpected_error_message
    from .base.context import context
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/base/context.py", line 21, in <module>
    from ..common.configuration import (Configuration, LoadError, MapParameter, PrimitiveParameter,
  File "/Users/user/anaconda/lib/python2.7/site-packages/conda/common/configuration.py", line 28, in <module>
    from enum import Enum, EnumMeta
ImportError: cannot import name EnumMeta

**Comment by adfoucart** [comment]
To anyone facing the same problem after uninstalling enum34 following bad StackOverflow advice, what solved the issue for me (on a Python 2.7 anaconda) what to install an older version of enum34:

`pip install enum34==1.1.2`

Afterwards, everything seems to work fine.
